src/kernels/backend_webgl.ts

```diff
@@ -815,8 +816,13 @@ export class MathBackendWebGL implements KernelBackend {
               transposeB, ENV.get('WEBGL_MATMUL_TS'),
               ENV.get('WEBGL_MATMUL_WPT'));
           break;
+        case 4:
+          program = new MatMulPackedProgramCSV4(
+              a.shape, [batch, outerShapeA, outerShapeB], transposeA,
+              transposeB);
+          break;
         default:
```

Comment:
if you want to try **Transposed input** , modify this to
```ts
        case 4:
          program = new MatMulPackedProgramCSV4(
              a.shape, [batch, outerShapeA, outerShapeB], transposeA,
              !transposeB);
          const output =
              this.makePackedTensor(program.outputShape, dtype) as Tensor3D;
          return this.compileAndRunCS<Tensor3D>(
              program, [a, b.transpose([0, 2, 1])], output);
```
But I didn't see an improvement in performance.
Maybe it‘s just for convenience.

> To get this idea implemented, we'll want to transpose one of the input matrices before starting the matrix-multiplication. This will save us a lot of trouble computing indices, as the K-sized dimension (which A and B share) will be the same dimension. We choose to transpose the B matrix.

Reply:
Does above change have impact on your experimental best configuration for V4?

Reply:
Performance becomes worse with this change.
36.01ms  -->  56.90ms

if revise code to 
```ts
        case 4:
          program = new MatMulPackedProgramCSV4(
              a.shape, [batch, outerShapeA, outerShapeB], transposeA,
              !transposeB, ENV.get('WEBGL_MATMUL_TS'),
              ENV.get('WEBGL_MATMUL_TSK'), ENV.get('WEBGL_MATMUL_WPT'));
          const output =
              this.makePackedTensor(program.outputShape, dtype) as Tensor3D;
          return this.compileAndRunCS<Tensor3D>(
              program, [a, b], output);
```
no `b.transpose([0, 2, 1])` (result is wrong)
it is 54.36ms